### PR TITLE
Merge prototyping into main, integrate Page_Parser class

### PR DESCRIPTION
**Pull request by chp2001**

### CODE REVIEW

File: src/page_parser.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
+# %pip install pdfminer.six
+import pdfminer
+import pdfminer.high_level as pdf_hl
+import pdfminer.layout as pdf_layout
```
Comment (themnsavage) [inline src/page_parser.py:4]: make sure to add pdfminer.six to requirements.txt file

File: src/page_parser.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+
+class Page_Parser:
+    def __init__(self, page: pdf_layout.LTPage):
+        self.page: pdf_layout.LTPage = page
+        self.unpacked: bool = False
+        self.bbox: tuple = page.bbox
+        self.width: int = page.width
+        self.height: int = page.height
+        self.grid = [
```
Comment (themnsavage) [inline src/page_parser.py:50]: if you want make these variables private and possibly add getter and setter functions if needed

File: src/page_parser.py [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+    for k, v in d2.items():
+        if k in d1:
+            if type(d1[k]) == dict:
+                recursive_dict_add(d1[k], v)
+            else:
+                d1[k] += v
+        else:
+            d1[k] = v
+
+
```
Comment (themnsavage) [inline src/page_parser.py:42]: could you possibly add these as private functions in the class

File: prototyping/page_0.png
Comment (rqualls) [inline prototyping/page_0.png:1]: very neat

File: src/pdf_parser.py [hunk 4]
```diff
@@ -1,9 +1,207 @@
...
+            "bd",
+            "it",
+            "lt",
+            "med",
+            "heavy",
+            "black",
+            "light",
+            "-",
+        ]
+
```
Comment (rqualls) [inline src/pdf_parser.py:134]: may want to put static data like this in a different file?

**Review by rqualls — APPROVED** [review]

**Pull request merged**